[Smart Accounts Kit] Add guide for EOA as a signer (#2575)
MetaMask/metamask-docs: smart-accounts-kit/guides/smart-accounts/signers/eoa-wallets.md

---
description: Learn how to use Externally Owned Account (EOA) with MetaMask Smart Accounts.
sidebar_label: EOA (e.g. MetaMask) 
keywords: [metamask, smart account, signer, metamask smart account]
---

# Use Externally Owned Account (EOA) with MetaMask Smart Accounts

> Externally Owned Account (EOA) are one of the most widely used wallet (e.g. MetaMask, Phantom, and more) whcih consists of public

Reviewer: ```suggestion
Externally Owned Accounts (EOAs) are one of the most widely used wallet (such as MetaMask, Phantom, and more). They consist of a public and private 
```

keypair where the private key is only known to the user. MetaMask Smart Accounts is a signer agnostic implementation that allows you
to use EOA as a signer for MetaMask Smart Accounts.

:::info

> This guide is targeted towards React and React-based frameworks. For Vue, see [Wagmi docs](https://wagmi.sh/vue/getting-started).

Reviewer: ```suggestion
This guide supports React and React-based frameworks. For Vue, see [Wagmi docs](https://wagmi.sh/vue/getting-started).
```

import { sepolia } from "viem/chains";

export const config = createConfig({
  chains: [sepolia],
  transports: {
    [sepolia.id]: http(),
  },
});
```

</TabItem>
</Tabs>

### 3. Create a smart account

Once the user has connected their wallet, use the [Wallet Client](https://viem.sh/docs/clients/wallet) from Wagmi as the signer to create a
MetaMask smart account.

```ts
import { Implementation, toMetaMaskSmartAccount } from "@metamask/smart-accounts-kit";
import { useAccount, usePublicClient, useWalletClient } from "wagmi";

const { address } = useAccount();
const publicClient = usePublicClient();
const { data: walletClient } = useWalletClient();

// Additional check to make sure the EOA wallet is connected
// and values are available.
if (!address || !walletClient || !publicClient ) {
  // Handle the error case
 }

const smartAccount = await toMetaMaskSmartAccount({
  client: publicClient,
  implementation: Implementation.Hybrid,
  deployParams: [address, [], [], []],
  deploySalt: "0x",
  signer: { walletClient },
});
```

## Next steps

- See how to use [MetaMask Embedded Wallets as a signer](./eoa-wallets.md) to make user onboarding journey easier.

> - See how to [send a user operations](../send-user-operation.md).

Reviewer 2: ```suggestion
- See how to [send a user operation](../send-user-operation.md).
```
